### PR TITLE
[#119] Fix upload button

### LINKED ISSUE #119: Upload button on profile settings page
**Describe the bug**
Currently, the upload button opens another file-upload window rather than submit the image, although this needs to be connected to the API regardless.

**To Reproduce**
1. Go to /profile/settings
2. Click on the 'EDIT AVATAR' button, then the 'Upload' button after uploading an image.

**Expected behavior**
Close the modal and display the new image as the avatar.

### PR DESCRIPTION
**Pull request by ejstuart**
# Description:
Updated backend to allow the uploading of images. Updated Profile Settings View Modal to fix issues with buttons not working, fixed Profile Settings Controller to call the modify users endpoint. Made changes to user model.

fixes #119

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [ ] My code has been commented
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

File: backend/server/controllers/user.js [hunk 1]
```diff
@@ -495,6 +495,7 @@ export const modifyUser = async (req, res) => {
       }
     }
   } catch (error) {
+    console.log(error)
```
Comment (scorpionknifes) [inline backend/server/controllers/user.js]:
```suggestion
```

**Review by scorpionknifes — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "Updog",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {}
+}
```
Comment (scorpionknifes) [inline package-lock.json]:
```suggestion
```
remove file

**Review by scorpionknifes — CHANGES_REQUESTED** [review]

**Review by scorpionknifes — APPROVED** [review]
LGTM

File: backend/server/controllers/image.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      contentType: mimetype,
+    },
+  })
+
+  const pipedStreams = stream.pipe(writeStream)
+
+  const result = new Promise((resolve, reject) => {
+    pipedStreams.on('finish', () => {
+      resolve({
+        imageURI: `${process.env.BACKEND_URL}images/${filename}`,
```
Comment (scorpionknifes) [inline backend/server/controllers/image.js:20]: we don't have backend url as .env. It fine, we just keep it there for now

**Review by f-fromont — APPROVED** [review]
Backend changes LGTM

**Pull request merged**